scripts/lib/k8s.sh

```diff
@@ -1,25 +1,59 @@
 #!/usr/bin/env bash
 
+CONTROLLER_TOOLS_VERSION="v0.4.0"
+
+# ensure_controller_gen checks that the `controller-gen` binary is available on
+# the host system and if it is, that it matches the exact version that we
+# require in order to standardize the YAML manifests for CRDs and Kubernetes
+# Roles.
+#
+# If the locally-installed controller-gen does not match the required version,
+# prints an error message asking the user to uninstall it.
 ensure_controller_gen() {
-  if ! is_installed controller-gen; then
-    # Need this version of controller-gen to allow dangerous types and float
-    # type support
-    go get sigs.k8s.io/controller-tools/cmd/controller-gen@4a903ddb7005459a7baf4777c67244a74c91083d
-  else
-    minimum_req_version="v0.3.1"
-    # Don't overide the existing version let the user decide.
-    if ! is_min_controller_gen_version "$minimum_req_version"; then
-        echo "Existing version of controller-gen "`controller-gen --version`", minimum required is $minimum_req_version"
-        exit 1
+    if ! is_installed controller-gen; then
+        __install_path=/usr/local/bin/controller-gen
```

Comment:
Users might have different Go install directories. Maybe let's use a more generic `GOBIN` path ? something like this should work
```
__install_path=$(go env GOBIN)/controller-gen
```

Reply:
Other things (like kind) are being installed into `/usr/local/bin` but OK :)

Reply:
Also, I personally don't have the GOBIN env var set... it's not set by default on Linux IIRC. You need to set it manually, even though `$GOPATH/bin` is the default for it:

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ go env GOBIN

jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ go env GOPATH
/home/jaypipes/go
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ ll $GOPATH
total 20
drwxr-xr-x  5 jaypipes jaypipes 4096 Aug 22  2019 ./
drwxr-xr-x 31 jaypipes jaypipes 4096 Oct  6 07:53 ../
drwxr-xr-x  2 jaypipes jaypipes 4096 Oct  6 07:53 bin/
drwxr-xr-x  5 jaypipes jaypipes 4096 Apr  5  2020 pkg/
drwxr-xr-x  8 jaypipes jaypipes 4096 Aug 25  2019 src/
```

Reply:
> Other things (like kind) are being installed into /usr/local/bin but OK :)

Ohh right! Forgot about kind case

> Also, I personally don't have the GOBIN env var set... it's not set by default on Linux IIRC. You need to set it manually, even though $GOPATH/bin is the default for it:

I always thought that `go env` doesn't use the OS env vars, i just found out that i have a `GOBIN` set in my `.bash_profile`... pre go1.11 habits :)